include/host/wasi/environ.h

```diff
@@ -831,6 +832,82 @@ class Environ {
     return {};
   }
 
+  WasiExpect<__wasi_fd_t> sockOpen(__wasi_address_family_t SysDomain,
+                                   __wasi_sock_type_t SockType) noexcept {
+
+    auto Node = getNodeOrNull(-1);
+
+    if (auto Res = VINode::sockOpen(FS, SysDomain, SockType); unlikely(!Res)) {
+      return WasiUnexpect(Res);
+    } else {
+      Node = std::move(*Res);
+    }
+
+    std::random_device Device;
```

Comment:
Same codes occured at `pathOpen`, should create a new member function for generate `NewFd`, and use same function at here and `pathOpen`.

Reply:
Impl as generateRandomFdToNode()